contrib/pg_tde/documentation/docs/functions.md

```diff
@@ -1,97 +1,254 @@
 # Functions
 
-The `pg_tde` extension provides the following functions:
+The `pg_tde` extension provides functions for managing different aspects of its operation:
 
-## pg_tde_add_key_provider_file
+## Permission management
 
-Creates a new key provider for the database using a local file.
+By default, `pg_tde` is restrictive. It doesn't allow any operations until permissions are granted to the user. Only superusers can run permission management functions to manage user permissions.
 
-This function is intended for development, and stores the keys unencrypted in the specified data file.
+Permissions are based on the normal `EXECUTE` permission on the functions provided by `pg_tde`. Superusers manage them using the `GRANT EXECUTE` and `REVOKE EXECUTE` commands.
 
-```
-SELECT pg_tde_add_key_provider_file('provider-name','/path/to/the/keyring/data.file');
-```
+The following functions are also provided for easier management of functionality groups:
 
-All parameters can be either strings, or JSON objects [referencing remote parameters](external-parameters.md).
+### Database local key management
+
+Use these functions to grant or revoke permissions to manage permissions for the current database. They enable or disable all functions related to the providers and keys on the current database:
+
+* `pg_tde_grant_local_key_management_to_role(role)`
+* `pg_tde_revoke_local_key_management_from_role(role)`
+
+### Global scope key management
+
+Use these functions to grant or revoke permissions to manage permissions for the global scope - the entire PostgreSQL instance. They enable or disable all functions related to the providers and keys for the global scope:
```

Comment:
We sometimes call it "PostgreSQL cluster" and sometimes "PostgreSQL instance". Is that intentional or should we standardize on one of them? I think the official is "cluster" even though that one might be confusing since cluster can also refer to a whole server cluster with a master and several replicas.